types/src/proto/transaction.proto

```diff
@@ -27,6 +27,13 @@ message SignedTransaction {
     bytes signed_txn = 5;
 }
 
+// A generic structure that represents a transaction, covering all three
+// possible variants: a signed user transaction, a write set transaction, or a
```

Comment:
we might not need to mention all these variants in comments. for example if people add another variant they might not remember to update this.